File: record-generator/processor/src/main/java/me/bristermitten/mittenlib/codegen/record/RecordSpecLike.java
package me.bristermitten.mittenlib.codegen.record;

import com.palantir.javapoet.ClassName;
import me.bristermitten.mittenlib.codegen.GenericTypeSpec;

public interface RecordSpecLike extends GenericTypeSpec {
    ClassName source();

    RecordConstructorSpec constructor();
}

File: record-generator/processor/src/main/scala/me/bristermitten/mittenlib/codegen/record/RecordSpecLike.scala
package me.bristermitten.mittenlib.codegen.record

import com.palantir.javapoet.ClassName
import me.bristermitten.mittenlib.codegen.GenericTypeSpec

trait RecordSpecLike extends GenericTypeSpec:
  def source: ClassName
  def constructor: RecordConstructorSpec
